Source/SceneEditor/SceneEditor.cpp

```diff
@@ -264,6 +251,311 @@ namespace PlatinumEngine{
+			Mesh* mesh = renderComponent->GetMesh();
+
+			// Make sure there is mesh
+			if(mesh != nullptr)
+			{
+				// loop all the vertices
+				for (int count = 0; count < mesh->indices.size(); count += 3)
+				{
+					//----------------------------------------------------------------//
+					// Find the world coordinate of the object (after transformation) //
+					//----------------------------------------------------------------//
+
+					Maths::Vec3 vertex0, vertex1, vertex2;
+
+					// check if the game object has transformation components
+					if (auto transformComponent = currentCheckingGameobject->GetComponent<TransformComponent>();
+							transformComponent != nullptr)
+					{
+						Maths::Mat4 translationMatrix, rotationMatrix, scaleMatrix;
+						translationMatrix.SetTranslationMatrix(transformComponent->position);
+						rotationMatrix.SetRotationMatrix(transformComponent->rotation.EulerAngles());
+						scaleMatrix.SetScaleMatrix(transformComponent->scale);
+
+						Maths::Vec4 temporaryMatrix;
+
+						// get the world coordinate of vertex 0
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 0]].position.x,
+												  mesh->vertices[mesh->indices[count + 0]].position.y,
+												  mesh->vertices[mesh->indices[count + 0]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex0 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+
+						// get the world coordinate of vertex 1
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 1]].position.x,
+												  mesh->vertices[mesh->indices[count + 1]].position.y,
+												  mesh->vertices[mesh->indices[count + 1]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex1 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+
+						// get the world coordinate of vertex 2
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 2]].position.x,
+												  mesh->vertices[mesh->indices[count + 2]].position.y,
+												  mesh->vertices[mesh->indices[count + 2]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex2 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+					}
+					else
+					{
+						vertex0 = mesh->vertices[mesh->indices[count + 0]].position;
+						vertex1 = mesh->vertices[mesh->indices[count + 1]].position;
+						vertex2 = mesh->vertices[mesh->indices[count + 2]].position;
+					}
+
+					//----------------------------------//
+					// Do ray and triangle intersection //
+					//----------------------------------//
+
+					// build the PCS using the triangle primitive
```

Comment:
What is PCS? Maybe add link for algorithm? This looks unfamiliar.

Reply:
Mentioned in intersection PowerPoint for fundamental modeling last semester.